helix-rest/src/main/java/org/apache/helix/rest/metadatastore/TrieRoutingData.java

```diff
@@ -37,12 +39,14 @@
 
   private final TrieNode _rootNode;
 
-  // TODO: THIS IS A TEMPORARY PLACEHOLDER. A proper constructor will be created, which will not
-  // take in a TrieNode; it instead initializes the rootNode and creates a trie based on
-  // some input data. The constructor is blocked by the implementation of RoutingDataAccessor, and
-  // will therefore be implemented later.
-  public TrieRoutingData(TrieNode rootNode) {
-    _rootNode = rootNode;
+  public TrieRoutingData(Map<String, List<String>> routingData) throws InvalidRoutingDataException {
+    if (isRootShardingKey(routingData)) {
```

Comment:
Is this necessary to special handling? following constructTrie logic will not handle the root sharding key?

Reply:
Yes it is necessary. TrieNodes' fields are mostly final, such as `_isLeaf`. Therefore, we always need to determine if the root node is leaf or not (and construct it) before any of the `constructTrie` code runs. Since it's always the first step, performing a check here is the cleanest way in my opinion.